ci: 👷 release,test-release,dep actions and pyproject.toml added

onuralpszr opened the pull request:
# Description

- Pyproject.toml (pypa, and well tested) 
- Dependabot added
- Release action added
- Test Release action added (rc,a,b)
- Tox ini and added as development dependencies  
- Github action for OSX (pytests) added
- Maestro version command added
- pyproject.toml exclude list updated
- License update from MIT to Apache
- Project urls are added.
- Keywords added.
- Exclude files list updated
- Changelog.md added

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/roboflow/multimodal-maestro?pullRequest=43) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/roboflow/multimodal-maestro?pullRequest=43) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/roboflow/multimodal-maestro?pullRequest=43) it.</sub>

onuralpszr commented:
Current test results 

```console
  py39: OK (66.17=setup[32.69]+cmd[33.48] seconds)
  py310: OK (57.87=setup[26.58]+cmd[31.29] seconds)
  py311: OK (55.35=setup[25.66]+cmd[29.69] seconds)
  py312: OK (61.36=setup[30.26]+cmd[31.10] seconds)
  congratulations :) (240.78 seconds)
```

It also confirms installation work properly

onuralpszr commented:
@SkalskiP  👋  I added most of the actions for maestro and I also removed setup.py and req.txt files and just use setuptools toml file "https://setuptools.pypa.io/" and I wrote action for osx and ubuntu to confirm all installations and tests are passed. 

I manually add flash-attention wheel for clean install because flash-attention doesn't have proper wheel files in pypi side it is only tarball which requires to be compile/build to be installed. But If we do release like this, it cannot be clean pip install maestro one-liner people either needs to have proper environment or install flash-attentation first along with torch to have complete install

onuralpszr commented:
##  Wheel Package Content list

adding 'maestro/__init__.py'
adding 'maestro/primitives.py'
adding 'maestro/visualizers.py'
adding 'maestro/cli/__init__.py'
adding 'maestro/cli/env.py'
adding 'maestro/cli/introspection.py'
adding 'maestro/cli/main.py'
adding 'maestro/cli/utils.py'
adding 'maestro/lmms/__init__.py'
adding 'maestro/lmms/gpt4.py'
adding 'maestro/markers/__init__.py'
adding 'maestro/markers/sam.py'
adding 'maestro/postprocessing/__init__.py'
adding 'maestro/postprocessing/mask.py'
adding 'maestro/postprocessing/text.py'
adding 'maestro/trainer/__init__.py'
adding 'maestro/trainer/common/__init__.py'
adding 'maestro/trainer/common/configuration/__init__.py'
adding 'maestro/trainer/common/configuration/env.py'
adding 'maestro/trainer/common/data_loaders/__init__.py'
adding 'maestro/trainer/common/data_loaders/datasets.py'
adding 'maestro/trainer/common/data_loaders/jsonl.py'
adding 'maestro/trainer/common/utils/__init__.py'
adding 'maestro/trainer/common/utils/file_system.py'
adding 'maestro/trainer/common/utils/leaderboard.py'
adding 'maestro/trainer/common/utils/metrics.py'
adding 'maestro/trainer/common/utils/reproducibility.py'
adding 'maestro/trainer/models/__init__.py'
adding 'maestro/trainer/models/florence_2/__init__.py'
adding 'maestro/trainer/models/florence_2/checkpoints.py'
adding 'maestro/trainer/models/florence_2/core.py'
adding 'maestro/trainer/models/florence_2/data_loading.py'
adding 'maestro/trainer/models/florence_2/entrypoint.py'
adding 'maestro/trainer/models/florence_2/metrics.py'
adding 'maestro/trainer/models/paligemma/__init__.py'
adding 'maestro/trainer/models/paligemma/entrypoint.py'
adding 'maestro/trainer/models/paligemma/training.py'
adding 'maestro-0.2.0rc3.dist-info/LICENSE'
adding 'maestro-0.2.0rc3.dist-info/METADATA'
adding 'maestro-0.2.0rc3.dist-info/WHEEL'
adding 'maestro-0.2.0rc3.dist-info/entry_points.txt'
adding 'maestro-0.2.0rc3.dist-info/top_level.txt'
adding 'maestro-0.2.0rc3.dist-info/RECORD'

PawelPeczek-Roboflow commented on .github/workflows/maestro-tests.yml:
```diff
@@ -0,0 +1,44 @@
+          pip install torch
+          if [[ "${{ matrix.os }}" == "ubuntu-latest" ]]; then
+            if [[ "${{ matrix.python-version }}" == "3.10" ]]; then
+              pip install https://github.com/Dao-AILab/flash-attention/releases/download/v2.6.3/flash_attn-2.6.3+cu123torch2.4cxx11abiTRUE-cp310-cp310-linux_x86_64.whl
```
will this break when torch is bumped to next version given flashattn is fixed to 2.4?

onuralpszr replied:
Yes, we need to update here, or I can lock pytorch in here and we can update when we wanted to as well. But source installation was breaking in action and I checked documentation of flash-attention it was asking other packages and cuda-toolkit to be present plus they also have certain version lock as you can see in pre-made as wheels
